src/app/options.h

```diff
@@ -28,46 +30,47 @@
  * Contact : chris@qbittorrent.org
  */
 
-#ifndef QINISETTINGS_H
-#define QINISETTINGS_H
+#ifndef APP_OPTIONS_H
+#define APP_OPTIONS_H
 
-#include <QSettings>
+#include <stdexcept>
 
-class QIniSettings : public QSettings {
-  Q_OBJECT
-  Q_DISABLE_COPY (QIniSettings)
+#include <QString>
+#include <QStringList>
 
-public:
-  QIniSettings(const QString &organization = "qBittorrent", const QString &application = "qBittorrent", QObject *parent = 0 ):
-#ifdef Q_OS_WIN
-      QSettings(QSettings::IniFormat, QSettings::UserScope, organization, application, parent)
+class QProcessEnvironment;
+struct QBtCommandLineParameters
+{
+    bool showHelp;
+#ifndef Q_OS_WIN
+    bool showVersion;
+#endif
+#ifndef DISABLE_GUI
+    bool noSplash;
 #else
-      QSettings(organization, application, parent)
+    bool shouldDaemonize;
 #endif
-  {
-
-  }
-
-  QIniSettings(const QString &fileName, Format format, QObject *parent = 0 ) : QSettings(fileName, format, parent) {
+    int webUiPort;
+    QString profileDir;
+    bool portableMode;
+    QString configurationName;
+    QStringList torrents;
+    QString unknownParameter;
 
-  }
+    QBtCommandLineParameters(const QProcessEnvironment&);
```

Comment:
Maybe remove `QProcessEnvironment` from constructor params and us `QProcessEnvironment::systemEnvironment()` in constructor body? You don't like yourself the extra things in the interface!